From 2d943c91595abf1a233aaa2e9d94782544ca1005 Mon Sep 17 00:00:00 2001
From: Gabriele M <moto.falcon.git@gmail.com>
Date: Sun, 26 Mar 2017 12:13:09 +0200
Subject: [PATCH 116/173] Always apply frame size reduction to panorama
 pictures

Panorama mode requires quite some memory, especially after the frame
num bump done with commit aa0733567c30 ("Make panorama able to go 270
degrees in landscape"). Some devices run out of memory while taking a
panorama picture and force close Snap before the picture is complete.
We have a config to reduce the memory requirements that reduces the
size of each frame, but it's applied only if ro.config.low_ram is
true. Bump the default value to 100 and always respect it. Devices
having ro.config.low_ram set to true will have to override this
config from their device tree.

BUGBASH-326

Change-Id: Ic6d24b17b2293adf8d715904c8c1874a4c624e99
---
 res/values/config.xml                               | 2 +-
 src/com/android/camera/WideAnglePanoramaModule.java | 8 +-------
 2 files changed, 2 insertions(+), 8 deletions(-)

diff --git a/res/values/config.xml b/res/values/config.xml
index 300c2d906..da52d90e1 100644
--- a/res/values/config.xml
+++ b/res/values/config.xml
@@ -22,7 +22,7 @@
 
     <!-- This value may be tweaked to save memory on low RAM devices. The value
          is the percentage of camera preview height/width to scale to. -->
-    <integer name="panorama_frame_size_reduction">90</integer>
+    <integer name="panorama_frame_size_reduction">100</integer>
 
     <!-- This value may be changed to true to enable the warped pano preview overlayed on top
          of the fullscreen pano preview. -->
diff --git a/src/com/android/camera/WideAnglePanoramaModule.java b/src/com/android/camera/WideAnglePanoramaModule.java
index 2e1e8fe08..b8d4775f1 100644
--- a/src/com/android/camera/WideAnglePanoramaModule.java
+++ b/src/com/android/camera/WideAnglePanoramaModule.java
@@ -16,7 +16,6 @@
 
 package com.android.camera;
 
-import android.app.ActivityManager;
 import android.content.ContentResolver;
 import android.content.Context;
 import android.content.Intent;
@@ -917,12 +916,7 @@ public class WideAnglePanoramaModule
             return;
         }
 
-        int perct = 100;
-        final ActivityManager am = (ActivityManager)
-                mActivity.getSystemService(Context.ACTIVITY_SERVICE);
-        if (am.isLowRamDevice()) {
-            perct = mActivity.getResources().getInteger(R.integer.panorama_frame_size_reduction);
-        }
+        int perct = mActivity.getResources().getInteger(R.integer.panorama_frame_size_reduction);
 
         int width = (mCameraPreviewWidth * perct) / 100;
         int height = (mCameraPreviewHeight * perct) / 100;
-- 
2.17.1

